src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Commands/SuppressAddReferenceCommand.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Input;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.Commands
+{
+    /// <summary>
+    /// Suppresses the "Add Reference..." command in specific dependencies tree context menus.
+    /// </summary>
+    [ProjectCommand(VSConstants.CMDSETID.StandardCommandSet2K_string, (long)VSConstants.VSStd2KCmdID.ADDREFERENCE)]
+    [AppliesTo(ProjectCapability.DependenciesTree)]
+    [Order(ProjectSystem.Order.Default)]
+    internal sealed class SuppressAddReferenceCommand : AbstractSingleNodeProjectCommand
+    {
+        // TODO once CPS inserts, use this field along with ContainsAny
+//      private static readonly ProjectTreeFlags s_hideAddReferenceForFlags = DependencyTreeFlags.DependenciesRootNode + DependencyTreeFlags.TargetNode;
+
+        protected override Task<CommandStatusResult> GetCommandStatusAsync(IProjectTree node, bool focused, string? commandText, CommandStatus progressiveStatus)
+        {
+//          if (node.Flags.ContainsAny(s_hideAddReferenceForFlags))
+            if (node.Flags.Contains(DependencyTreeFlags.DependenciesRootNode) || node.Flags.Contains(DependencyTreeFlags.TargetNode))
+            {
+                return GetCommandStatusResult.Suppressed;
```

Comment:
This should be handled, enabled but hidden and only on the context menu (I'll try and find the flag). Otherwise, this prevents Project -> Add Reference from working if the dependency node is selected, with no replacement.

Reply:
I think the flag to do this OLECMDF_DEFHIDEONCTXTMENU which CPS doesn't yet expose.

Reply:
Addressed in d52300585 (will test once VS build is ready).